### DISCUSSION TITLE
SQLite3

### DISCUSSION THREAD

**Discussion by tinygiant98** [damms005/devdb-vscode#34] [Q&A] [answered]
Just installed the preview version and attempted to open a current sqlite3 db file on my computer (via the `SQLite Database File Picker` button), but DevDB refuses to recognize it as a valid file and won't open it.  Does DevDB understand the newer versions of sqlite or am I missing something obvious?  Thanks.

**Comment by damms005** [comment]
Sorry you're experiencing this issue. Please run this command on your terminal and reply with the output:

```
sqlite3 /path/to/your/your_database_file.db "PRAGMA integrity_check;"
```

**Reply by tinygiant98** [reply]
`ok`

**Reply by tinygiant98** [reply]
If it matters, all of my files are of extension `.sqlite3`, but still resulted in `ok` with that check.  Also tried renaming it to db since the files are exactly the same and it would then show up in the filepicker filtered window, but still would not open.  With my other `.sqlite3` files, I'd have to filter to `All Files`.

**Reply by damms005** [reply]
The file extension doesn't really matter.

Your terminal output also suggests that your database file is good.

Can you help me with a screen recording of this issue please? Also, it'll be helpful if you can send a copy of your file. Don't forget to truncate the table data if you have non-test data like sensitive or private data in there.

**Reply by tinygiant98** [reply]
Thanks for the replies.  A screen recording won't do much good here.  Here's the sequence of events:
- Click on the "SQLite Database File Picker" button
- Change the file filter to "All Files" so I can see my `.sqlite3` files
- Select my database file and click on the `Open SQLite File` button
- File Picker dialog closes and ... nothing, just returned to vscode

Attached is the very basic `test.sqlite3` database I'm trying to open (sorry for the zip, github didn't allow direct uploads of db files).  There was no sensitive info in it anyway (it's a test case), but I deleted pretty much everything out of it except table definitions and few rows of data to drop the file size dramatically.
[test.zip](https://github.com/damms005/devdb-vscode/files/13838264/test.zip)

**Reply by damms005** [reply]
This is super weird because your db opens for me. I simply unzipped it and loaded it. See recording below:


https://github.com/damms005/devdb-vscode/assets/9839355/5a6a5090-1534-424c-8643-1d1083d281ec


Any error logged into the developer console? (you can open it with `Ctrl + Shift + P` and search for "developer console" then toggle it on)

**Reply by tinygiant98** [reply, marked as answer]
Nope, absolutely zero output on the developer console.  Thanks for your help so far, no need to spend a bunch of time on this.  I'll try re-installing and give it another shot.

**Reply by damms005** [reply]
Nice! Please let me know how that goes.
